packages/frontend-api/src/Model/Customer/User/LoginType/CustomerUserLoginTypeRepository.php

```diff
@@ -20,16 +20,50 @@ public function __construct(
     /**
      * @param \Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser $customerUser
      * @param string $loginType
-     * @return bool
+     * @return \Shopsys\FrontendApiBundle\Model\Customer\User\LoginType\CustomerUserLoginType|null
      */
-    public function existsByCustomerUserAndType(
+    public function findByCustomerUserAndType(
         CustomerUser $customerUser,
         string $loginType,
-    ): bool {
+    ): ?CustomerUserLoginType {
         return $this->entityManager->getRepository(CustomerUserLoginType::class)
-            ->count([
+            ->findOneBy([
                 'customerUser' => $customerUser,
                 'loginType' => $loginType,
-            ]) > 0;
+            ]);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser $customerUser
+     * @return \Shopsys\FrontendApiBundle\Model\Customer\User\LoginType\CustomerUserLoginType
+     */
+    public function getMostRecentLoginType(CustomerUser $customerUser): CustomerUserLoginType
+    {
+        return $this->entityManager->getRepository(CustomerUserLoginType::class)
+            ->createQueryBuilder('cult')
+            ->where('cult.customerUser = :customerUser')
+            ->orderBy('cult.lastLoggedInAt', 'DESC')
+            ->setParameter('customerUser', $customerUser)
+            ->setMaxResults(1)
+            ->getQuery()
+            ->getSingleResult();
```

Comment:
It can throw `NoResultException`. Shouldn't it be covered?

Reply:
good catch, thanks! :ok_hand: fixed in https://github.com/shopsys/shopsys/pull/3276/commits/540765ae69145e8dfba6a41345f655964d5c684f